articles/azure/azs-how-setup-monitoring.md

```diff
@@ -42,164 +42,387 @@ To complete the steps in this article, you must have appropriate access to a sub
+    Get-AzNetworkSecurityGroup -Name $NetworkSecurityGroupName -ResourceGroupName $ResourceGroupName | New-AzNetworkSecurityRuleConfig -Name "Port443-Rule" -Description "Allow port 443" -Access "Allow" -Protocol "TCP" -Direction "Inbound" -Priority 100 -DestinationPortRange 443 -SourceAddressPrefix "*" -SourcePortRange "*" -DestinationAddressPrefix "*" | Set-AzNetworkSecurityGroup
+
+    # Determine if VM is Windows or Linux based
+    if ($VM.OsProfile.WindowsConfiguration) {
+        $DependencyAgentType = "DependencyAgentWindows"
+        $DependencyAgentVersion = "9.10"
+        $MonitoringAgentType = "MicrosoftMonitoringAgent"
+        $MonitoringAgentVersion = "1.0"
+    }
+    else {
+        $DependencyAgentType = "DependencyAgentLinux"
+        $DependencyAgentVersion = "9.10"
+        $MonitoringAgentType = "OmsAgentForLinux"
+        $MonitoringAgentVersion = "1.13"
+    }
 
     # Deploy DependencyAgent extension
     Set-AzVMExtension -ExtensionName "DependencyAgent" `
-    -ResourceGroupName $VM.ResourceGroupName `
-    -VMName $VM.Name `
-    -Publisher "Microsoft.Azure.Monitoring.DependencyAgent" `
-    -ExtensionType "DependencyAgentLinux" `
-    -TypeHandlerVersion 9.7 `
-    -Location $Location `
-    -Verbose
+                      -ResourceGroupName $VM.ResourceGroupName `
+                      -VMName $VM.Name `
+                      -Location $Location `
+                      -Publisher "Microsoft.Azure.Monitoring.DependencyAgent" `
+                      -ExtensionType $DependencyAgentType `
+                      -TypeHandlerVersion $DependencyAgentVersion `
+                      -Verbose
 
     # Deploy Microsoft.EnterpriseCloud.Monitoring extension
     Set-AzVMExtension -ExtensionName "Microsoft.EnterpriseCloud.Monitoring" `
-    -ResourceGroupName $VM.ResourceGroupName `
-    -VMName $VM.Name `
-    -Publisher "Microsoft.EnterpriseCloud.Monitoring" `
-    -ExtensionType "OmsAgentForLinux" `
-    -TypeHandlerVersion 1.12 `
-    -SettingString $PublicSettings `
-    -ProtectedSettingString $ProtectedSettings `
-    -Location $Location `
-    -Verbose</code></pre>
-
-6. Continue to step 8.
+                      -ResourceGroupName $VM.ResourceGroupName `
+                      -VMName $VM.Name `
+                      -Location $Location `
+                      -Publisher "Microsoft.EnterpriseCloud.Monitoring" `
+                      -ExtensionType $MonitoringAgentType `
+                      -TypeHandlerVersion $MonitoringAgentVersion `
+                      -SettingString $PublicSettings `
+                      -ProtectedSettingString $ProtectedSettings `
+                      -Verbose</code></pre>
 
 ***
 
-8. In public Azure, on the top left, click the hamburger icon and select **Monitor**.
+# [Portal](#tab/tabid-1)
```

Comment:
Should there be a heading here to explain what these steps are for?

Reply:
Agreed, added a header.

Reply:
LGTM